.github/workflows/build.yml

```diff
@@ -0,0 +1,282 @@
+                    compression-level: 9
+
+            -   name: Temporary Artifacts Cleanup
+                run: rm -rf ./artifacts
+
+    integration-tests:
+        name: Integration Tests
+        runs-on: ubuntu-latest
+        needs: [ checkstyle, spotbugs ]
+        steps:
+            -   name: Checkout Repository
+                uses: actions/checkout@v4
+
+            -   name: Setup Java 21 LTS
+                uses: actions/setup-java@v4
+                with:
+                    java-version: 21
+                    distribution: 'temurin'
+
+            -   name: Start Docker Dependencies
+                env:
+                    PEM111_AMQP_BROKER: "amqp://activemq:5672"
+                    PEM111_AMQP_PORT: "5672"
+                    PEM111_ITK_EXTERNAL_CONFIGURATION_URL: "http://wiremock:8080/configuration/ods-dos"
+                    LOG_LEVEL: DEBUG
+                run: |
+                    docker network create 111network
+                    docker compose build
+                    docker compose up activemq wiremock --detach
+                working-directory: ./docker
+
+            -   name: Execute Integration Tests
+                run: ./gradlew integrationTest
+                working-directory: ./service
+
+            -   name: Dump Docker Logs
+                if: always()
+                run: |
+                    chmod +x dump_docker_logs.sh
+                    ./dump_docker_logs.sh
+                working-directory: ./scripts
+                shell: bash
+
+            -   name: Collect Artifacts
+                if: always()
+                run: |
+                    mkdir -p artifacts
+                    cp -r ./service/build/reports ./artifacts
+                    cp -r ./scripts/logs ./artifacts
+
+            -   name: Upload Artifacts
+                uses: actions/upload-artifact@v4
+                if: always()
+                with:
+                    name: 'Integration Test Reports & Docker Logs'
+                    path: ./artifacts/**
+                    compression-level: 9
+
+            -   name: Stop Docker Dependencies
+                if: always()
+                run: |
+                    docker compose down --rmi=local --volumes --remove-orphans
+                    docker compose rm
+                    docker network rm 111network
+                working-directory: ./docker
+
+            -   name: Temporary Artifacts Cleanup
+                run: rm -rf ./artifacts
+
+    generate-build-id:
+        name: Generate Build ID
+        runs-on: ubuntu-latest
+        needs: [unit-tests, integration-tests]
+        outputs:
+            build-id: ${{ steps.generate.outputs.buildId }}
+        steps:
+            -   name: Checkout Repository
+                uses: actions/checkout@v4
+
+            -   id: generate
+                working-directory: ./scripts
+                shell: bash
+                env:
+                    GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Why do you need to inject the GH_TOKEN ?

Reply:
I've removed it